src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -1048,6 +1053,17 @@ bool ConditionMgr::IsObjectMeetingVendorItemConditions(uint32 creatureId, uint32
     return true;
 }
 
+bool ConditionMgr::IsObjectMeetingAreaTriggerConditions(uint32 areaTriggerId, bool isServerSide, WorldObject* object) const
+{
+    ConditionEntriesByAreaTriggerIdMap::const_iterator itr = AreaTriggerConditionContainerStore.find({ areaTriggerId, isServerSide });
```

Comment:
This is exactly where you DO use `auto`, or preferably `Trinity::Containers::MapGetValuePtr` since this is a map lookup